test/auth_oidc/test_auth_oidc.py

```diff
@@ -252,104 +316,164 @@ def test_reauthenticate_succeeds(self):
                 "find",
             ],
         )
+        # Assert that the list of command succeeded events is [find].
         self.assertEqual(succeeded_events, ["find"])
+        # Assert that a find operation failed once during the command execution.
         self.assertEqual(failed_events, ["find"])
-
-        # Assert that the request callback has been called twice.
-        self.assertEqual(self.request_called, 2)
+        # Close the client.
         client.close()
 
-    def test_reauthenticate_succeeds_no_refresh(self):
+    def test_4_2_reauthenticate_succeeds_no_refresh(self):
+        # Create a default OIDC client with a human callback that does not return a refresh token.
         cb = self.create_request_cb()
 
-        def request_cb(*args, **kwargs):
-            result = cb(*args, **kwargs)
-            del result["refresh_token"]
-            return result
+        class CustomRequest(OIDCCallback):
+            def fetch(self, *args, **kwargs):
+                result = cb.fetch(*args, **kwargs)
+                result.refresh_token = None
+                return result
 
-        # Create a client with the callback.
-        props: Dict = {"request_token_callback": request_cb}
-        client = MongoClient(self.uri_single, authmechanismproperties=props)
+        client = self.create_client(request_cb=CustomRequest())
 
-        # Perform a find operation.
+        # Perform a find operation that succeeds.
         client.test.test.find_one()
 
-        # Assert that the request callback has been called once.
+        # Assert that the human callback has been called once.
         self.assertEqual(self.request_called, 1)
 
+        # Force a reauthenication using a fail point.
         with self.fail_point(
             {
                 "mode": {"times": 1},
                 "data": {"failCommands": ["find"], "errorCode": 391},
             }
         ):
-            # Perform a find operation.
+            # Perform a find operation that succeeds.
             client.test.test.find_one()
 
-        # Assert that the request callback has been called twice.
+        # Assert that the human callback has been called twice.
         self.assertEqual(self.request_called, 2)
+        # Close the client.
         client.close()
 
-    def test_reauthenticate_succeeds_after_refresh_fails(self):
-        # Create request callback that returns valid credentials.
-        request_cb = self.create_request_cb()
-
-        # Create a client with the callback.
-        props: Dict = {"request_token_callback": request_cb}
-        client = MongoClient(self.uri_single, authmechanismproperties=props)
+    def test_4_3_reauthenticate_succeeds_after_refresh_fails(self):
+        # Create a default OIDC client.
```

Comment:
NIT: I liked the comment from the earlier usage here. Default made me have to do a double-take. 
```suggestion
        # Create a client with a human callback that returns a valid token.
```